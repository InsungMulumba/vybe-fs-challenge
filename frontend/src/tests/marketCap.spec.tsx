/**
 * @jest-environment jsdom
 */

import * as renderer from "react-test-renderer";
import MarketCap from "../charts/marketCap";

jest.mock("react-apexcharts", () => ({
  __esModule: true,
  default: () => <div />,
}));

const testData = [
  {
    name: "Token1",
    address: "EPedfgdfgdfsg5GGon7jmAWKVUHuux1Tpz",
    marketCap: 70004.38,
  },
  {
    name: "Token2",
    address: "a1sFFFwFtw2kb48X77W9xq8hBZ5kVfdd321W",
    marketCap: 55555.6,
  },
  {
    name: "Token3",
    address: "ki87nWEUx17Y4AggKG444BYerwr1Pd8wZ4cp",
    marketCap: 66666.92,
  },
  {
    name: "Token4",
    address: "8LzFRE3B890023TKYz9Drzqnpgee3SGarqM",
    marketCap: 44447.9,
  },
  {
    name: "Token5",
    address: "CBB55111SCo9jkCejsZfz8Ec8nH7TEKSnvwT6XK6",
    marketCap: 99987.85,
  },
];

describe("Market Cap Component", () => {
  it("renders Pie Chart correctly", () => {
    const tree = renderer.create(<MarketCap data={testData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
