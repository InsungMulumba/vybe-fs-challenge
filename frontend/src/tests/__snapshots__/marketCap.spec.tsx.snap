// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market Cap Component renders Pie Chart correctly 1`] = `<div />`;
